### PR TITLE
added config option to register mbeans

### PR DESCRIPTION
**Pull request by Deepak-initcron**
register-mbeans configure option allow to setRegisterMbeans option. Which can we use to setup jmx monitoring for hikaricp.

### CODE REVIEW

File: src/hikari_cp/core.clj [hunk 1]
```diff
@@ -76,7 +77,8 @@
    :maximum-pool-size  IntGte1
    :adapter            AdaptersList
    (s/optional-key :leak-detection-threshold) IntGte2000
-   s/Keyword           s/Any})
+   s/Keyword           s/Any
+   :register-mbeans    s/Bool})
```
Comment (tomekw) [inline src/hikari_cp/core.clj]: Please move this line before `s/Keyword           s/Any`.

**Comment by tomekw** [comment]
@Deepak-initcron could you also update the `README.md` and the `CHANGELOG.md` under the `unreleased` section?

**Comment by tomekw** [comment]
Thanks! :)

**Comment by Deepak-initcron** [comment]
done..

**Comment by tomekw** [comment]
@Deepak-initcron thanks!

**Pull request merged**